modules/spring-config/src/main/java/org/mule/runtime/config/spring/MuleArtifactContext.java

```diff
@@ -177,6 +170,13 @@ protected XmlConfigurationDocumentLoader newXmlConfigurationDocumentLoader() {
     return new XmlConfigurationDocumentLoader();
   }
 
+  private XmlApplicationParser createApplicationParser() {
+    ExtensionManager extensionManager = muleContext.getExtensionManager();
+    DslResolvingContext context =
+        extensionManager != null ? DslResolvingContext.getDefault(extensionManager.getExtensions()) : null;
+    return new XmlApplicationParser(new XmlApplicationServiceRegistry(serviceRegistry, context));
```

Comment:
Disregard my previous comment on this, I was talking about something completely different. 
In this case, if context is null then it'll contribute with an empty set of extension models and that's it. I'll change this to avoid using the null value.